specification/src/chapter2.adoc

```diff
@@ -2,9 +2,600 @@
+
+==== Authorized software 
+
+Running unauthorized software can compromise the security state of the system. Two complementary processes can be used to authorize software:
+
+* Measuring 
+* Verification
+
+A measurement is a cryptographic fingerprint of the launch state of a software component, such as a running hash of its memory contents and launch state.
+
+Verification compares an actual measurement to an expected measurement from a signed authorization by a trusted signer. Verification requires a provisioned or securely discovered list of trusted signer(s). Depending on use case, a signed authorization can be part of a software image. Or it can be delivered or discovered separately as part of an authorization protocol.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All isolated software that has to be trusted by other software on the same system MUST be measured and verified.
+|===
+
+Verification ensures only software from trusted parties is installed.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All isolated software that provides services to remote reliant parties MUST be at least measured.
+|===
+
+Software that is at least measured can be attested by remote reliant parties. To ensure supply chain integrity it is recommended that it is also verified locally before installation.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Software authorization MUST be rooted in immutable boot code.
+|===
+
+Immutable boot code forms part of the immutable hardware security platform identified by the hardware security platform identity and is the start of a local trust chain on a system. It is typically ROM code, or locked on-chip flash.
+
+If a system implements a hardware root of trust, then the HW root of trust hosts the immutable boot code. Otherwise the immutable boot code is typically the first code that executes on an application processor immediately after reset. 
+
+==== Secure updates
+
+Over time, any non-immutable component may need updates to address vulnerabilities or functionality improvements. An update can concern software/firmware, microcode, or any other updatable element on a system.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| All software on a system which is not immutable MUST be updatable.
+|===
+
+Immutable software includes at least immutable boot code. Some trusted subsystems can also include immutable software to meet specific security certification requirements defined by a governing body. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| An update of a software component SHOULD restart all other dependant software components.
```

Comment:
this will be limiting capabilities that can securely update a layered TCB component without restarting dependent software component. (it is expected in these cases, the dependent SW components explicitly opt-in to updates, and the attestation reports that opt-in apriori along with the creation security version and the current security version of the TCB.

Reply:
This is the catch-all. Live-updates are covered just below. Are you ok with that?

Reply:
I have updated the "Authorized software" section above to prepare the ground. Please review.

Will tweak the update section tomorrow.

Reply:
Pushed an update to the system update section as well. Please review.